@codex please fix

```
> dspace@3.0.0 build
> node scripts/run-astro-build.mjs

 ✓ tests/ingressTlsConfig.test.ts (11 tests) 7ms
 ✓ tests/questFormExistingQuests.test.ts (1 test) 20ms
 ✓ tests/preparePrScriptSteps.test.ts (4 tests) 6ms
 ✓ tests/postinstall.test.ts (4 tests) 13ms
07:16:27 [types] Generated 832ms
07:16:27 [build] output: "server"
07:16:27 [build] directory: /home/runner/work/dspace/dspace/frontend/dist/
07:16:27 [build] adapter: @astrojs/node
07:16:27 [build] Collecting build info...
07:16:27 [build] ✓ Completed in 943ms.
07:16:27 [build] Building server entrypoints...
 ✓ tests/packageManager.test.ts (4 tests) 10ms
 ✓ frontend/tests/questStats.test.ts (3 tests) 7ms
 ✓ tests/offlineFirstDocSync.test.ts (1 test) 7ms
 ✓ scripts/tests/fsChecks.test.ts (6 tests) 12ms
 ✓ frontend/src/components/__tests__/menuActive.spec.ts (5 tests) 6ms
 ✓ frontend/src/pages/docs/__tests__/index.spec.ts (2 tests) 167ms
 ✓ frontend/tests/docsRagComparison.test.ts (5 tests) 8ms
 ✓ scripts/tests/imageReferences.test.ts (1 test) 51ms
 ✓ tests/questPrerequisites.test.ts (1 test) 29ms
 ✓ tests/hardeningQuality.test.ts (2 tests) 83ms
 ✓ frontend/src/pages/chat/svelte/__tests__/Integrations.spec.ts (1 test) 128ms
 ✓ tests/titlesLogic.test.ts (2 tests) 10ms
 ✓ tests/roadmapChecklist.test.ts (1 test) 8ms
 ✓ tests/questRewardGrantSeparation.test.ts (1 test) 171ms
 ✓ tests/dCarbonOffsetDoc.test.ts (1 test) 9ms
 ✓ tests/openAI.persona.test.ts (1 test) 696ms
   ✓ GPT5Chat persona integration > injects persona-specific prompt and welcome message  694ms
 ✓ tests/itemValidation.test.ts (2 tests) 92ms
 ✓ tests/processQuality.test.ts (3 tests) 110ms
 ✓ tests/purgeClientStateRetry.test.ts (2 tests) 11ms
 ✓ frontend/tests/item-requirements.test.ts (3 tests) 5ms
 ✓ tests/strings.test.ts (9 tests) 7ms
 ✓ frontend/tests/questListManifest.test.ts (2 tests) 79ms
 ✓ tests/changelogContributorsGuide.test.ts (1 test) 6ms
 ✓ tests/scoreQuestOpenAI.test.ts (2 tests) 14ms
 ✓ scripts/tests/validateStagedQuests.test.ts (3 tests) 33ms
07:16:48 [build] ✓ Completed in 20.35s.

 building client (vite) 
 ✓ tests/generateItemDependencies.test.ts (3 tests) 95ms
 ✓ tests/navigateWithRetry.test.ts (2 tests) 11ms
 ✓ tests/noConflictMarkers.test.ts (1 test) 1208ms
   ✓ repository sanity ‑ no merge-conflict markers > contains no <<<<<<< or >>>>>>> markers  1206ms
 ✓ tests/k8sEnvironmentOverlays.test.ts (2 tests) 6ms
 ✓ scripts/tests/questDependencies.test.ts (2 tests) 6ms
 ✓ tests/svelteResolution.test.ts (4 tests) 462ms
 ✓ tests/architectureNotesAccessibilityBaseline.test.ts (2 tests) 4ms
 ✓ tests/questDependencyReferences.test.ts (1 test) 33ms
 ✓ tests/questIdPathConsistency.test.ts (1 test) 27ms
 ✓ tests/svelteSubpathResolver.test.ts (2 tests) 6ms
 ✓ tests/serviceWorkerConfigCache.test.ts (1 test) 7ms
 ✓ tests/noDuplicateItemIds.test.ts (1 test) 5ms
 ✓ frontend/tests/docsMarkdownSafety.test.ts (1 test) 15ms
 ✓ tests/questReliabilityCoverage.test.ts (1 test) 172ms
 ✓ tests/v3ReleaseStateDoc.test.ts (3 tests) 8ms
 ✓ frontend/tests/questListClassifier.test.ts (3 tests) 6ms
 ✓ tests/changelogQuestBranchingNote.test.ts (1 test) 13ms
 ✓ tests/metricsEndpoint.test.ts (3 tests) 25ms
 ✓ tests/docsSearchFeatures.test.ts (2 tests) 8ms
 ✓ tests/questProcessRecoveryPaths.test.ts (1 test) 148ms
 ✓ tests/questsPolishPromptSections.test.ts (3 tests) 5ms
 ✓ tests/builtinQuestSchema.test.ts (1 test) 84ms
 ✓ tests/githubTokenPersistence.test.ts (1 test) 11ms
 ✓ tests/spellcheckConfig.test.ts (1 test) 13ms
 ✓ frontend/src/components/__tests__/QuestLinkChips.spec.ts (2 tests) 229ms
07:17:05 [build] Rearranging server assets...
07:17:05 [build] Server built in 38.55s
07:17:05 [build] Complete!
 ✓ tests/buildOutputValidation.test.ts (1 test) 45782ms
   ✓ build output validation > should not contain /src/scripts/ paths in built files  45779ms
 ✓ tests/pnpmFrontendFilter.test.ts (1 test) 17ms
 ✓ tests/scanSecretsScript.test.ts (2 tests) 102ms
 ✓ tests/helmChartReleaseVersion.test.ts (1 test) 7ms
 ✓ tests/changelogDocsAICoPilots.test.ts (1 test) 6ms
 ✓ tests/compactItemListHelpers.test.ts (2 tests) 6ms
 ✓ tests/customContentLoadTest.test.ts (2 tests) 1370ms
   ✓ runCustomContentLoadTest > uses default count when omitted  1348ms
 ✓ tests/userJourneysCoverage.test.ts (1 test) 5ms
 ✓ tests/measureEcSolutionRegression.test.ts (1 test) 5ms
 ✓ tests/changelogDocsContentUpdate.test.ts (1 test) 8ms
 ✓ scripts/tests/detectZeroTests.test.ts (5 tests) 4ms
 ✓ tests/customQuestLoader.test.ts (1 test) 17ms
 ✓ tests/raspiSecretsDocs.test.ts (2 tests) 5ms
 ✓ scripts/tests/preparePrScripts.test.ts (1 test) 4ms
 ✓ tests/chartResourcesDefaults.test.ts (1 test) 19ms
 ✓ tests/inject-ids.test.ts (1 test) 6ms
 ✓ tests/helmChartGracefulShutdown.test.ts (2 tests) 4ms
 ✓ tests/cheatsAvailability.test.ts (5 tests) 6ms
 ✓ tests/changelogCloudSync.test.ts (1 test) 5ms
 ✓ tests/docsRagDrift.test.ts (2 tests) 462ms
   ✓ docs RAG drift intent > forces the v3 release state doc for v2-only drift queries  445ms
 ✓ tests/changelogProcessRequirementsNote.test.ts (1 test) 5ms
 ✓ tests/markdownEncodingFix.test.ts (3 tests) 4ms
 ✓ tests/helmChartPersistence.test.ts (2 tests) 4ms
 ✓ frontend/src/components/__tests__/RemainingTime.spec.ts (2 tests) 55ms
 ✓ tests/quest-title-unique.test.ts (3 tests) 4ms
 ✓ tests/questCopyRegression.test.ts (1 test) 4ms
 ✓ scripts/tests/validateItem.test.ts (2 tests) 12ms
 ✓ tests/chatPersonas.test.ts (1 test) 178ms
 ✓ tests/workflowPnpmCache.test.ts (3 tests) 6ms
 ✓ tests/changelogNpcChatNote.test.ts (1 test) 4ms
 ✓ frontend/tests/chatHints.test.ts (4 tests) 7ms
 ✓ tests/changelogProcessReservation.test.ts (1 test) 6ms
 ✓ tests/questTreesDoc.test.ts (3 tests) 5ms
 ✓ tests/deployEnvFluxDocs.test.ts (1 test) 5ms
 ✓ tests/networkPolicyObservability.test.ts (2 tests) 4ms
 ✓ tests/roadmapDocLinks.test.ts (1 test) 5ms
 ✓ tests/backupScript.test.ts (2 tests) 161ms
 ✓ tests/guildsDocs.test.ts (2 tests) 6ms
 ✓ scripts/tests/validateQuest.test.ts (2 tests) 32ms
 ✓ tests/seedCustomContent.test.ts (1 test) 32ms
 ✓ tests/menuToolbox.test.ts (2 tests) 6ms
 ✓ tests/roadmapPastYearsComplete.test.ts (1 test) 5ms
 ✓ tests/fix-markdown-image-paths.test.ts (1 test) 5ms
 ✓ tests/promptsCodexLinks.test.ts (1 test) 4ms
 ✓ tests/itemSelectorLocators.test.ts (1 test) 5ms
 ✓ tests/helmChartProbes.test.ts (1 test) 5ms
 ✓ tests/customContentItemsStore.test.ts (1 test) 14ms
 ✓ tests/userJourneysDoc.test.ts (2 tests) 5ms
 ✓ tests/buildWorkflowMultiArch.test.ts (2 tests) 6ms
 ✓ tests/processHardening.test.ts (1 test) 8ms
 ✓ tests/cacheVersionSync.test.ts (1 test) 12ms
 ✓ tests/questSchemaValidation.test.ts (2 tests) 8ms
 ✓ tests/dbBenchmarkCli.test.ts (1 test) 310ms
   ✓ db-benchmark CLI > accepts --count option  308ms
 ✓ tests/canvasDependencyAlignment.test.ts (1 test) 4ms
 ✓ tests/changelogDocsReleaseChecklist.test.ts (1 test) 3ms
 ✓ tests/playwrightConfigBrowsers.test.ts (1 test) 4ms
 ✓ frontend/src/components/__tests__/SearchBar.spec.ts (1 test) 65ms
 ✓ tests/customContentBackupImporterAccessibility.test.ts (1 test) 5ms
 ✓ tests/license.test.ts (3 tests) 5ms
 ✓ tests/questGuidelinesEditorDoc.test.ts (1 test) 3ms
 ✓ tests/ubiQuestLeaderboardCopy.test.ts (1 test) 5ms
 ✓ frontend/tests/playwrightConfigHost.test.ts (1 test) 6ms
 ✓ tests/roadmap10xQuests.test.ts (1 test) 10ms
 ✓ tests/questGraphValidation.test.ts (2 tests) 329ms
 ✓ tests/rocketsDocsContent.test.ts (1 test) 7ms
 ✓ tests/solarBasicsIcons.test.ts (5 tests) 6ms
 ✓ scripts/tests/lintStagedItemValidation.test.ts (1 test) 5ms
 ✓ scripts/tests/lintStagedQuestValidation.test.ts (1 test) 5ms
 ✓ tests/rocketsDocsPromise.test.ts (1 test) 5ms
 ✓ tests/fsChecksInvalidEncoding.test.ts (1 test) 8ms
 ✓ tests/roadmapSuborbitalLaunch.test.ts (1 test) 5ms
 ✓ tests/roadmapCustomQuests.test.ts (1 test) 6ms
 ✓ tests/changelogNotes.test.ts (1 test) 4ms
 ✓ tests/testingDocsPromises.test.ts (1 test) 4ms
 ✓ frontend/src/components/__tests__/compactItemListHelpers.spec.ts (1 test) 3ms
 ✓ tests/k3sChecklistComplete.test.ts (1 test) 4ms
 ✓ tests/docsFAQ.test.ts (1 test) 3ms
 ✓ tests/roadmapPromises.test.ts (1 test) 8ms
 ✓ tests/solarDocWind.test.ts (1 test) 4ms
 ✓ tests/docsPromptsOutages.test.ts (2 tests) 5ms
 ✓ tests/menu-active.test.ts (1 test) 6ms
 ✓ tests/sanitization.test.ts (1 test) 19ms
 ✓ tests/itemSelectorTouch.test.ts (2 tests) 5ms
 ✓ tests/metricsFallback.test.ts (1 test) 37ms
 ✓ tests/process-form-mobile.test.ts (1 test) 6ms
 ✓ tests/questCountIncrease.test.ts (1 test) 14ms
 ✓ tests/quest-form-mobile.test.ts (1 test) 5ms
 ✓ tests/item-form-mobile.test.ts (1 test) 4ms
 ✓ tests/menu-leaderboard.test.ts (1 test) 4ms
 ✓ tests/titlesDoc.test.ts (1 test) 4ms
 ✓ tests/togglePreviewId.test.ts (3 tests) 4ms
 ✓ scripts/tests/questQuality.test.ts (1 test) 9ms
 ✓ tests/npmTestCiScript.test.ts (1 test) 8ms
 ✓ tests/menu-settings.test.ts (1 test) 7ms
 ✓ tests/achievementsCatalog.test.ts (1 test) 6ms
 ✓ tests/mergePlanChecklist.test.ts (1 test) 4ms
 ✓ frontend/__tests__/ShoppingForm.test.js (1 test) 3ms
 ✓ tests/menu-stats.test.ts (1 test) 11ms
 ✓ tests/profileTitles.test.ts (1 test) 6ms
 ✓ frontend/tests/menu.test.ts (1 test) 5ms
 ✓ tests/menuTitles.test.ts (1 test) 6ms
 ✓ frontend/tests/duration.test.ts (2 tests) 4ms
 ✓ tests/pnpmfile.test.ts (1 test) 5ms
 ✓ tests/domExceptionGlobal.test.ts (1 test) 5ms
 ✓ tests/domexception-global.test.ts (1 test) 5ms
 ✓ tests/domexception-globals.test.ts (1 test) 4ms
 ✓ tests/sysadminQuest.test.ts (1 test) 4ms
 ✓ tests/shopIndex.test.ts (1 test) 4ms
 ✓ tests/eslintPluginPresence.test.ts (1 test) 5ms
 ✓ frontend/tests/gameStateFlush.test.ts (1 test) 6ms
 ✓ tests/checkoutFetchDepth.test.ts (1 test) 5ms
 ✓ scripts/tests/prettierFormatting.test.ts (1 test) 29120ms
   ✓ Prettier formatting > frontend sources are formatted  29118ms
 ✓ tests/npmTestRunsAllSuites.test.ts (1 test) 5ms
 ✓ tests/astroConfigLogLevel.test.ts (1 test) 5ms
 ✓ tests/prismSvelte.test.ts (1 test) 16ms
 ✓ scripts/tests/dbBenchmarkScript.test.ts (1 test) 5ms
 ✓ tests/progressionBalance.test.ts (1 test) 29ms
 ✓ tests/domException.test.ts (1 test) 5ms
 ✓ tests/indexeddb.server.test.ts (1 test) 273ms
 ✓ frontend/tests/sanity.test.ts (1 test) 4ms

⎯⎯⎯⎯⎯⎯ Failed Suites 8 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  tests/docsRagMetaFreshness.test.ts [ tests/docsRagMetaFreshness.test.ts ]
Error: Failed to resolve import "../frontend/src/generated/rag/docs_meta.json" from "tests/docsRagMetaFreshness.test.ts". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/tests/docsRagMetaFreshness.test.ts:2:21
  1  |  import { describe, expect, it } from "vitest";
  2  |  import docsMeta from "../frontend/src/generated/rag/docs_meta.json";
     |                        ^
  3  |  import { getDocsRagComparison, getDocsRagMismatchWarning } from "../frontend/src/utils/docsRag.js";
  4  |  describe("docs RAG metadata freshness", () => {
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/21]⎯

 FAIL  frontend/tests/chatHallucinationContract.test.ts [ frontend/tests/chatHallucinationContract.test.ts ]
Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯��⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/21]⎯

 FAIL  frontend/tests/docsRag.test.ts [ frontend/tests/docsRag.test.ts ]
Error: Failed to resolve import "../src/generated/rag/docs_meta.json" from "frontend/tests/docsRag.test.ts". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/tests/docsRag.test.ts:2:21
  1  |  import { describe, expect, it, vi } from "vitest";
  2  |  import docsMeta from "../src/generated/rag/docs_meta.json";
     |                        ^
  3  |  import { rankDocsResults, searchDocsRag } from "../src/utils/docsRag.js";
  4  |  vi.setConfig({ testTimeout: 12e4, hookTimeout: 12e4 });
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/21]⎯

 FAIL  frontend/tests/openAI.test.ts [ frontend/tests/openAI.test.ts ]
Error: Failed to resolve import "../generated/build_meta.json" from "frontend/src/utils/buildInfo.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/buildInfo.js:1:24
  1  |  import buildMeta from '../generated/build_meta.json';
     |                         ^
  2  |  
  3  |  const readViteGitSha = () => {
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/21]⎯

 FAIL  frontend/tests/openAIChatDocsRagComparison.test.ts [ frontend/tests/openAIChatDocsRagComparison.test.ts ]
Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock
 ❯ frontend/src/utils/openAI.js:5:31
      3| import { buildDchatKnowledgePack } from './dchatKnowledge.js';
      4| import { mergeSources } from './contextSources.js';
      5| import { searchDocsRag } from './docsRag.js';
       |                               ^
      6| import { npcPersonas } from '../data/npcPersonas.js';
      7| import OpenAI from 'openai';

Caused by: Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/21]⎯

 FAIL  frontend/tests/openAIChatErrorMessages.test.ts [ frontend/tests/openAIChatErrorMessages.test.ts ]
Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock

 ❯ frontend/src/pages/chat/svelte/OpenAIChat.svelte:5:31
      4|         defaultOpenAIErrorMessage,
      5|         describeOpenAIError,
      6|         buildChatPrompt,
       |  ^
      7|         CHAT_PROMPT_VERSION,
      8|         getPlayerStateSummary,

Caused by: Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/21]⎯

 FAIL  frontend/tests/promptScaffoldingGuardrails.test.ts [ frontend/tests/promptScaffoldingGuardrails.test.ts ]
Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
 Test Files  11 failed | 283 passed (294)
      Tests  13 failed | 937 passed (950)
   Start at  07:15:13
   Duration  170.07s (transform 11.99s, setup 11.62s, collect 39.31s, tests 122.32s, environment 200.42s, prepare 38.48s)

     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/21]⎯

 FAIL  frontend/src/pages/chat/svelte/__tests__/ChatDebugBuildInfo.spec.ts [ frontend/src/pages/chat/svelte/__tests__/ChatDebugBuildInfo.spec.ts ]
Error: Failed to resolve import "../generated/build_meta.json" from "frontend/src/utils/buildInfo.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/buildInfo.js:1:24
  1  |  import buildMeta from '../generated/build_meta.json';
     |                         ^
  2  |  
  3  |  const readViteGitSha = () => {
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[8/21]⎯


⎯⎯⎯⎯⎯⎯ Failed Tests 13 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > uses the non-reasoning model and includes guardrails in the system prompt
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > replaces suspicious precision without citations with a safe fallback
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > keeps suspicious precision when inline citations are present
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > keeps suspicious precision when a /docs link is present
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > dedupes fallback models and retries with the next unique model
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > answers completed quests using PlayerState when present
 FAIL  tests/chatNonReasoningRegression.test.ts > chat non-reasoning regression probes > requests /gamesaves export when PlayerState is missing
 FAIL  tests/chatPromptVersionStamp.test.ts > chat prompt version stamp > includes the prompt version header and guardrail lines
 FAIL  tests/chatPromptVersionStamp.test.ts > chat prompt version stamp > stamps prompt version with the build SHA when available
 FAIL  tests/chatPromptVersionStamp.test.ts > chat prompt version stamp > emits v3:<shortsha> in production mode without a build SHA
 FAIL  tests/chatPromptVersionStamp.test.ts > chat prompt version stamp > does not emit a missing prompt version in the system message
 FAIL  tests/gpt5ChatResponses.test.ts > gpt-5 chat responses integration > calls the Responses API with knowledge context and returns text output
 FAIL  tests/gpt5ChatResponses.test.ts > gpt-5 chat responses integration > seeds a welcome assistant message when no conversation history exists
Error: Failed to resolve import "../generated/build_meta.json" from "frontend/src/utils/buildInfo.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/buildInfo.js:1:24
  1  |  import buildMeta from '../generated/build_meta.json';
     |                         ^
  2  |  
  3  |  const readViteGitSha = () => {
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[9/21]⎯


Error: Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock
 ❯ frontend/src/utils/openAI.js:5:31

Caused by: Caused by: Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27



Error: Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock
 ❯ frontend/src/pages/chat/svelte/OpenAIChat.svelte:5:31

Caused by: Caused by: Error: Failed to resolve import "../generated/rag/docs_chunks.json" from "frontend/src/utils/docsRag.js". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/runner/work/dspace/dspace/frontend/src/utils/docsRag.js:57:24
  55 |          docsRagPromise = (async () => {
  56 |              const [chunksModule, indexModule, metaModule] = await Promise.all([
  57 |                  import('../generated/rag/docs_chunks.json'),
     |                         ^
  58 |                  import('../generated/rag/docs_index.json'),
  59 |                  import('../generated/rag/docs_meta.json'),
 ❯ TransformPluginContext._formatError node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49258:41
 ❯ TransformPluginContext.error node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49253:16
 ❯ normalizeUrl node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64307:23
 ❯ node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64439:39
 ❯ TransformPluginContext.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:64366:7
 ❯ PluginContainer.transform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:49099:18
 ❯ loadAndTransform node_modules/.pnpm/vite@5.4.21_@types+node@24.3.0/node_modules/vite/dist/node/chunks/dep-BK3b2jBa.js:51978:27


 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
     "pretest:root": "node frontend/scripts/build-processes.mjs",
     "dev": "cd frontend && npm run dev",
     "test": "npm run test:pr",
-    "test:root": "vitest run --config vitest.config.mts --testTimeout 20000 --maxWorkers=1",
+    "test:root": "npm run build:meta && npm run build:docs-rag && vitest run --config vitest.config.mts --testTimeout 20000 --maxWorkers=1",
     "coverage": "vitest run --config vitest.config.mts --coverage",
     "test:watch": "cd frontend && npm run test:watch",
     "test:e2e": "cd frontend && npm run test:e2e",
@@ -56,7 +56,8 @@
     "preview": "cd frontend && npm run preview",
     "lint:a11y": "cd frontend && npm run lint:a11y",
     "dev:legacy": "npm run dev",
-    "playwright:install": "npm --prefix frontend run playwright:install"
+    "playwright:install": "npm --prefix frontend run playwright:install",
+    "build:meta": "node scripts/write-build-meta.mjs"
   },
   "devDependencies": {
     "@jest/globals": "^29.5.0",

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "format": "cd frontend && npm run format",
     "format:check": "cd frontend && npm run format:check",
     "format:check:frontend-canary": "npx prettier --check frontend/__tests__/offlineWorkerRegistration.test.js",
-    "type-check": "tsc --noEmit",
+    "type-check": "npm run build:meta && npm run build:docs-rag && tsc --noEmit",
     "build:docs-rag": "node scripts/build-docs-rag-index.mjs",
     "build": "node scripts/build-with-sha.mjs",
     "verify:chat-build-stamp": "node scripts/verify-chat-build-stamp.mjs",

--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "dev": "cd frontend && npm run dev",
     "test": "npm run test:pr",
     "test:root": "npm run build:meta && npm run build:docs-rag && vitest run --config vitest.config.mts --testTimeout 20000 --maxWorkers=1",
-    "coverage": "vitest run --config vitest.config.mts --coverage",
+    "coverage": "npm run build:meta && npm run build:docs-rag && vitest run --config vitest.config.mts --coverage",
     "test:watch": "cd frontend && npm run test:watch",
     "test:e2e": "cd frontend && npm run test:e2e",
     "test:e2e:coverage": "cd frontend && npm run test:e2e:coverage",

